### ISSUE TITLE
Loading topologies from OpenMM sometimes scrambles atom order

### ISSUE THREAD

**Issue by j-wags** [openforcefield/openff-interchange#934]
**Description**

Today, a user reported that creating an Interchange from an OpenMM System+Topology with many lipids would lead to the simulation exploding within about 10 1-2 fs timesteps. 

The user used a PDB containing a protein+lipids (no water or solvent) parameterized using OpenMM's PDBFile and then ForceField class (`openmm_ForceField(*["amber/ff14SB.xml", "amber/tip3p_standard.xml", "amber/lipid17.xml"])`). They also sent a docked small molecule through OpenFF (`openff-2.1.1`) and used `create_interchange`, and then combined the interchange objects. When minimizing, the initial energy was about 1E10 kJ/mol, but it minimized down to about -5E5. The resulting simulation object exploded within a few timesteps both when using `to_openmm` and `to_openmm_simulation`. Viewing the trajectory, the initial coordinates looked good, however the user noted that some carbons in the lipids were labeled as hydrogens in the trajectory PDB.

We believe that this is (at least in part) due to a case where the networkx `connected_components_subgraphs` method will return subgraphs with the nodes not sorted in ascending order of node ID (indeed, it makes no guarantee that it would ever do so anyway). This might mangle the order of particles in a molecule, which may in turn affect the forces, since the 1-4s and exclusions may be calculated from the bond graph. 

**Reproduction**

To reproduce the atom ordering bug, use this PDB:

```
HETATM 4256  O   HOH F   1      50.109  -6.533 -26.604  1.00  0.00           O  
HETATM 4257  H1  HOH F   1      49.167  -6.695 -26.653  1.00  0.00           H  
HETATM 4258  H2  HOH F   1      50.334  -6.698 -25.688  1.00  0.00           H  
TER    4259      HOH F   1
HETATM 4260  H1  HOH G   1      29.239 -14.646 -38.831  1.00  0.00           H  
HETATM 4261  O   HOH G   1      29.363 -13.719 -39.032  1.00  0.00           O  
HETATM 4262  H2  HOH G   1      29.790 -13.360 -38.254  1.00  0.00           H  
TER    4263      HOH G   1
HETATM 4264  H1  HOH H   1      37.415 -14.008 -22.620  1.00  0.00           H  
HETATM 4265  H2  HOH H   1      37.340 -14.092 -23.571  1.00  0.00           H  
HETATM 4266  O   HOH H   1      36.605 -13.574 -22.353  1.00  0.00           O  
TER    4267      HOH H   1
HETATM 4268  H1  HOH I   1      33.346  -9.551 -40.029  1.00  0.00           H  
HETATM 4269  O   HOH I   1      32.484  -9.259 -39.735  1.00  0.00           O  
HETATM 4270  H2  HOH I   1      33.180  -9.980 -40.869  1.00  0.00           H  
TER    4271      HOH I   1
```

and this code:

```python
from openmm.app import PDBFile

pdb = PDBFile("copied_reordered_water.pdb")
from openff.interchange.components.toolkit import _simple_topology_from_openmm
top = _simple_topology_from_openmm(pdb.topology)
for mol in top.molecules:
    print("next mol")
    for atom in mol.atoms:
        print(atom.to_dict())
```

> next mol
> {'metadata': {}, 'atomic_number': 8}
> {'metadata': {}, 'atomic_number': 1}
> {'metadata': {}, 'atomic_number': 1}
> next mol
> {'metadata': {}, 'atomic_number': 1}
> {'metadata': {}, 'atomic_number': 8}
> {'metadata': {}, 'atomic_number': 1}
> next mol
> {'metadata': {}, 'atomic_number': 8} # The problem is here, this water should go H-H-O
> {'metadata': {}, 'atomic_number': 1}
> {'metadata': {}, 'atomic_number': 1}
> next mol
> {'metadata': {}, 'atomic_number': 1}
> {'metadata': {}, 'atomic_number': 8}
> {'metadata': {}, 'atomic_number': 1}

**Comment by mattwthompson** [comment]
I believe we fixed this in #933, but some more testing would be useful

**Comment by mattwthompson** [comment]
Closing this as the original issue is resolved - still would be useful to add more tests to this code path
